param/__init__.py

```diff
@@ -775,22 +773,37 @@ def __init__(self,default=(0,0),length=None,**params):
             self.length = length
 
         self._check(default)
-        Parameter.__init__(self,default=default,**params)
+        super(Tuple,self).__init__(default=default,**params)
+
 
     def _check(self,val):
+        if val is None and self.allow_None:
+            return
+
         if not isinstance(val,tuple):
             raise ValueError("NumericTuple '%s' only takes a tuple value."%self._attrib_name)
 
         if not len(val)==self.length:
             raise ValueError("%s: tuple is not of the correct length (%d instead of %d)." %
                              (self._attrib_name,len(val),self.length))
-        for n in val:
-            if not _is_number(n):
-                raise ValueError("%s: tuple element is not numeric: %s." % (self._attrib_name,str(n)))
+
 
     def __set__(self,obj,val):
         self._check(val)
-        super(NumericTuple,self).__set__(obj,val)
+        super(Tuple,self).__set__(obj,val)
+
+
+
+class NumericTuple(Tuple):
+    """A numeric tuple Parameter (e.g. (4.5,7.6,3)) with a fixed tuple length."""
+
+    __slots__ = []
```

Comment:
On balance, I'd vote for leaving out the empty slots declarations, because we aren't using them consistently, and people will wonder (a) what it's there for (which they can find in Python documentation, but there's still probably no point in them going to look it up), and (b) why they aren't everywhere.  Seems better to be consistent; the fact that we use slots shouldn't matter to most users of Param, and it won't hurt if they add empty slots themselves, but we don't need to implicitly suggest that they do so by putting a few around to confuse them...